sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2265,4 +2266,15 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val df = spark.range(1).select($"id", new Column(Uuid()))
     checkAnswer(df, df.collect())
   }
+
+  test("SPARK-24051: using the same alias can produce incorrect result") {
```

Comment:
This test case failed without your changes?

Reply:
yes, without the change the result is:
```
+---+---+---+
|  a|  b|  n|
+---+---+---+
|  1|  0|  2|
|  2|  0|  2|
|  3|  0|  1|
+---+---+---+
```